components/pixels/lib.rs

```diff
@@ -157,22 +175,73 @@ pub fn load_from_memory(buffer: &[u8], cors_status: CorsStatus) -> Option<Image>
             debug!("{}", msg);
             None
         },
-        Ok(_) => match image::load_from_memory(buffer) {
-            Ok(image) => {
-                let mut rgba = image.into_rgba8();
-                rgba8_byte_swap_colors_inplace(&mut rgba);
-                Some(Image {
-                    width: rgba.width(),
-                    height: rgba.height(),
-                    format: PixelFormat::BGRA8,
-                    bytes: IpcSharedMemory::from_bytes(&rgba),
-                    id: None,
-                    cors_status,
-                })
+        Ok(format) => match format {
+            ImageFormat::Gif => {
+                let decoder = GifDecoder::new(Cursor::new(buffer));
+                let mut width = 0;
+                let mut height = 0;
+                match decoder {
+                    Ok(dec) => {
+                        let mut frames = vec![];
+                        let frames_result = dec.into_frames();
+                        for frame_result in frames_result {
+                            match frame_result {
+                                Ok(mut f) => {
```

Comment:
This could also be an early return. Do not use abbreviations (unless very common like html) for variable names. Also, I think when printing an error, given some context about where the error is happening:
```suggestion
   let frame_result = match {
       Ok(frame_result) => frame_result,
       Err(error) => {
           warn!("Could not decode gif: {error}");
           return None;
       }
```

Reply:
For the early retur here, the concern is that, we may have multiple good frames before encountering a bad frame, and maybe we should store these good frames in the `Image`. I encounter this situation when trying to pass wpt test: `/html/semantics/embedded-content/the-img-element/image-loading-lazy.html`, where it fail when loading the blob image.
minimal test case should be: 
```
<html>
<body>
<img  src="data:image/png;base64,R0lGODlhDAAMAKIFAF5LAP/zxAAAANyuAP/gaP///wAAAAAAACH5BAEAAAUALAAAAAAMAAwAAAMlWLPcGjDKFYi9lxKBOaGcF35DhWHamZUW0K4mAbiwWtuf0uxFAgA">
</body>
</html>
```
which will show the below image:
![image-blob](https://github.com/user-attachments/assets/a0dba7bd-6855-4542-ae59-d923413950be)